src/continuous_agg.c

```diff
@@ -989,3 +999,97 @@ ts_continuous_agg_find_integer_now_func_by_materialization_id(int32 mat_htid)
 	}
 	return par_dim;
 }
+
+TS_FUNCTION_INFO_V1(ts_continuous_agg_watermark);
+
+/*
+ * Get the watermark for a real-time aggregation query on a continuous
+ * aggregate.
+ *
+ * The watermark determines where the materialization ends for a continuous
+ * aggregate. It is used by real-time aggregation as the threshold between the
+ * materialized data and real-time data in the UNION query.
+ *
+ * The watermark is defined as the end of the last (highest) bucket in the
+ * materialized hypertable of a continuous aggregate.
+ *
+ * The materialized hypertable ID is given as input argument.
+ */
+Datum
+ts_continuous_agg_watermark(PG_FUNCTION_ARGS)
+{
+	int32 hyper_id = PG_GETARG_INT32(0);
```

Comment:
Is it safe to call this before checking `PG_ARGISNULL`?

Reply:
Yes, that's my understanding. It might be a garbage value, however. But this function is also marked `STRICT` so the NULL check is a bit redundant. But I think we decided to do NULL checks anyway, if I am not mistaken.